### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by piesieomane**
In this Pull Request the following list below was implemented;
- Decorate a class.
- Use the Decorate design pattern.
- Create an `Interface` `nameable.
- Prepare `base Decorator`

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Nameable
```
Comment (Ismailco) [inline base_decorator.rb]: - [ ] The class CapitalizeDecorator should inherit from BaseDecorator, Not Nameable class, kindly fix this so your Decorator works as expected

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @piesieomane, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_decorator.rb (hunk 1 above)
Comment (piesieomane) [reply base_decorator.rb]: Thank you very much

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @piesieomane , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**